torch/export/export_schema.py

```diff
@@ -0,0 +1,326 @@
+        # SYMINT = auto()    # !!! Can Symint[] be an arguement? there are symint[] in native_function.yaml
+
+        # GRAPH = auto()      # !!! Consider how to handle condition op, which need to pass in a graph for the branch
+        # GRAPHS = auto()      # !!! What about list of graphs? Do we need this?
+
+        # !!! Following types doesn't have a list version in native_function.yaml
+        SCALAR_TYPE = auto()
+        MEMORY_FORMAT = auto()
+        LAYOUT = auto()
+        DEVICE = auto()
+
+    # Declaration of which type the value field is storing
+    type: ArgumentType
+
+    val: Union[
+        TensorArgument,
+        # List[TensorArgument],
+        Scalar,
+        # List[Scalar],
+        bool,
+        List[bool],
+        int,
+        List[int],
+        float,
+        List[float],
+        str,
+        # List[str],
+        # SymInt,
+        # List[SymInt],
+        # Graph,
+        # List[Graph],
+        ScalarType,
+        MemoryFormat,
+        Layout,
+        Device,
+    ]
+
+@dataclass
+class KeywordArgument:
+    key: str
+    vale: Argument
+
+# !!! How to model optional fields? Is it an operator schema annotation, or an argument type?
+#     Tensor?
+#     Scalar?
+#     ScalarType?
+#     bool?
+#     int?
+#     int[]?
+#     float[]?
+#     SymInt[]?
+#     MemoryFormat?
+#     Layout?
+#     Device?
+
+
+################################################################################
+# Following section is the defining the schema of serializing a concrete tensor
+
+# TensorMeta is a decription of a tensor, without the actual data (,effectively maps to FakeTensor)
+# TensorMeta has multliple uses
+#   1. Represent the property of a concrete tensor backed by a storage
+#     - This is used in the serialization of a concrete tensor, e.g. model weight
+#     - In this case, sizes and strides must be concrete ints, and cannot be symbolic
+#     - stride and storage_offset have to used to correctly reconstruct the tensor from the storage
+#   2. Represent the property of a virtual tensor (see IValue below)
+#     - In this case, sizes and strides can be either concrete ints or symbolic ints.
+#     - device/strides/storage_offset/layout/memory_format are tied to pytorch's implementation.
+#       These are faithful capture of tensor's detail in pytorch's executions during tracing
+#       However, it's up to downstream system on how to utilized these fields
+#       In another word, these feilds are suggestive, rather than mandatory.
+
+class TensorMeta:
+    dtype: ScalarType
+    sizes: List[SymInt]
+
+    # needed for training
+    requires_grad: bool
+
+    # !!! see description above, there are subtle difference on how these fields should be interpreted
+    device: Device
+    strides: List[SymInt]
+    storage_offset: int
+    layout: Layout
+    memory_format: MemoryFormat
+
+
+@dataclass
+class Buffer:
```

Comment:
This seems is the only thing we need;
Although, need size; 
we dont need type here because TensorMeta have it, do we need endianness though?

Reply:
1. Correct. We don't need dtype here. The buffer should be interpreted according to the TensorMeta. 
2. Yes, we need to define endianness for this buffer. It's arguable whether we need to it configurable though. I think we should just pick one and document is in the spec. 
3. In most serialization libraries, bytes are intrinsic type, so we most likely don't need an explicit size here.

Reply:
+1 on the endianness thing. It should not be configurable

Reply:
I would use big-endian, unless anyone have strong opinion on this.

Reply:
I think 1 and 3 make sense... it's what the runtime schema and any other schema I've seen do.
Endianness also should be opinionated for this.